### PR TITLE
Emacs dune mode

### PR DESCRIPTION
**Pull request by Chris00**

### CODE REVIEW

**Comment by rgrinberg** [comment]
I think you ended up bringing back the skeletons which we agreed to remove for now.

**Comment by Chris00** [comment]
> I think you ended up bringing back the skeletons which we agreed to remove for now.

Did we?  I think hey should be removed at the same time they can be generated.  Unless it can be done right away, removing them know means it will be uncomfortable to switch to this repo for my use — there are quite a few things to improve still and it is better if I use the mode.

**Comment by rgrinberg** [comment]
I see. Then I'm ok with keeping the skeletons for now.

**Comment by Chris00** [comment]
I have added a dune file to generate autoloads and install the files.

**Comment by Chris00** [comment]
One probably wants to create a "dune-emacs" package that depends on conf-emacs.

**Comment by rgrinberg** [comment]
> One probably wants to create a "dune-emacs" package that depends on conf-emacs.

What are the benefits? Users should really not install emacs packages via opam anyway. Also, these .el files don't really need emacs to be on the system just so they're installed anyway.

File: editor-integration/emacs/dune [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ (targets dune-site-file.el)
+ (deps    dune-mode.el dune-flymake.el)
+ (action  (progn
+           (write-file "%{targets}"
+            ";;; %{targets} --- Automatically extracted autoloads.
+;;; Code:
+(add-to-list 'load-path
+  (or (file-name-directory load-file-name) (car load-path)))
+
")
```
Comment (rgrinberg) [inline editor-integration/emacs/dune]: What is this quote?

**Comment by Chris00** [comment]
On 14 September 2018 at 06:43 GMT, Rudi Grinberg wrote:
>
> +;;; Code:
> +(add-to-list 'load-path
> +  (or (file-name-directory load-file-name) (car load-path)))
> +")
>
> What is this quote?

This is so the user only has to (load "path/to/dune-site-file" t) to set up the library.

**Comment by Chris00** [comment]
> What are the benefits? Users should really not install emacs packages via opam anyway. Also, these .el files don't really need emacs to be on the system just so they're installed anyway.

I don't but I think that quite a few people use the package user-setup (883 installs last month) and, as Dune will be part of the platform, it would be good that user-setup takes care of installing dune-mode.

**Comment by rgrinberg** [comment]
> I don't but I think that quite a few people use the package user-setup (883 installs last month) and, as Dune will be part of the platform, it would be good that user-setup takes care of installing dune-mode.

Sure, but there's no need to complicate the setup with `dune-emacs`, `conf-emacs` to achieve that. Users can just install dune and have these files available exactly as they'd expect.

**Comment by Chris00** [comment]
Removed the controversial commits.

**Review by rgrinberg — APPROVED** [review]
Thanks. I'm not against this extra stuff if it's for developers only, but creating another package just for this is not worth the effort. Especially if package management in Emacs is not really done through opam anyway.

**Comment by rgrinberg** [comment]
@chris00 yeah that would be preferable

**Comment by Chris00** [comment]
Done (and a bit more...).

**Comment by rgrinberg** [comment]
Thanks for the hard work Christophe. I've added you as a code owner to the .el files.

**Pull request merged**